drape/harfbuzz_shaping.cpp

```diff
@@ -0,0 +1,218 @@
+#include "drape/harfbuzz_shaping.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include <string>
+
+#include <unicode/ubidi.h>    // ubidi_open, ubidi_setPara
+#include <unicode/uscript.h>  // UScriptCode
+#include <utf8/unchecked.h>
+
+namespace harfbuzz_shaping
+{
+namespace
+{
+constexpr size_t kMaxScriptsForACharacter = 32;
+
+// Writes the script and the script extensions of the Unicode codepoint.
+// Returns the number of written scripts.
+size_t GetScriptExtensions(char32_t codepoint, UScriptCode * scripts)
```

Comment:
nit: Here and below you can use more C++ way with std::array or std::vector without assigning vars by pointer :)